src/hyperion/parameters/beamline_parameters.py

```diff
@@ -30,22 +32,55 @@ def from_lines(cls, config_lines: list[str]):
             if len(param) == 2
         ]
         for i, (param, value) in enumerate(config_pairs):
-            if value == "Yes":
-                config_pairs[i] = (config_pairs[i][0], True)
-            elif value == "No":
-                config_pairs[i] = (config_pairs[i][0], False)
-            elif value in BEAMLINE_PARAMETER_KEYWORDS:
-                pass
-            else:
-                config_pairs[i] = (config_pairs[i][0], float(config_pairs[i][1]))
+            try:
+                # BEAMLINE_PARAMETER_KEYWORDS effectively raw string but whitespace removed
+                if value not in BEAMLINE_PARAMETER_KEYWORDS:
+                    config_pairs[i] = (
+                        config_pairs[i][0],
+                        cls.parse_value(config_pairs[i][1]),
+                    )
+            except Exception as e:
+                LOGGER.warning(f"Unable to parse {file_name} line {i}: {e}")
+
         ob.params = dict(config_pairs)
         return ob
 
     @classmethod
     def from_file(cls, path: str):
         with open(path) as f:
             config_lines = f.readlines()
-        return cls.from_lines(config_lines)
+        return cls.from_lines(path, config_lines)
+
+    @classmethod
+    def parse_value(cls, value: str):
+        if value[0] == "[":
+            return cls.parse_list(value[1:].strip())
+        else:
+            return cls.parse_list_element(value)
+
+    @classmethod
+    def parse_list_element(cls, value: str):
+        if value == "Yes":
+            return True
+        elif value == "No":
+            return False
+        else:
+            return float(value)
+
+    @classmethod
+    def parse_list(cls, value: str):
+        list_output = []
+        remaining = value.strip()
+        i = 0
+        while (i := remaining.find(",")) != -1:
+            list_output.append(cls.parse_list_element(remaining[:i]))
+            remaining = remaining[i + 1 :].lstrip()
+        if (i := remaining.find("]")) != -1:
+            list_output.append(cls.parse_list_element(remaining[:i]))
+            remaining = remaining[i + 1 :].lstrip()
+        else:
+            raise Exception("Missing closing ']' in list expression")
```

Comment:
Could: it might be better to have a more specific exception here, e.g. `ValueError`, or define a new one - and catch it above, so that we don't throw away any other exceptions, but since it's on a per-line basis I'm not too worried about it

Reply:
Might be nice to have a test for this exception too